### ISSUE TITLE
Encoding for object/array subtypes (e.g raster-cube)?

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-processes#299]
We have some subtypes such as `raster-cube`, `vector-cube` and `labeled-array` that don't have an encoding in JSON and with the subtypes we work around that. For examples and tests (see #204) it would be good to have a JSON encoding defined for them.

For example, `raster-cube` could be encoded as follows:

```js
{
  "cube:dimensions": {...}, // see STAC spec for details
  "cube:data": [...] // multi dimensional array
}
```

For example, `labeled-array` could be encoded as follows:
```js
[
 {
   label: '2010-01',
   value: 123
  },
 {
   label: '2010-02',
   value: 321
  },
  ...
]
```

**Comment by soxofaan** [comment]
>  `"cube:data": [...] // multi dimensional array`
>  Nevertheless, in raster-cube you'd have the issue that some values can't be represented (i.e. nan and +/-inf)

Working with multidimension JSON arrays is probably fine for small examples and tests, but for larger cases (and to also cover nan/inf/...)  you might want the option to have the cube's data as a binary blob, encoded as a string in hex or base64 for example.

**Comment by m-mohr** [comment]
Yes, this is mostly meant for small chunks and tests. Nevertheless, the nan and +/-inf issue is somewhat a blocker for me, but we may still allow the limited JSON as a simple way of doing things and find an alternative for these special cases. Binary is also not straightforward though as you've mentioned with e.g. data types etc. If we need the data type details in the metadata though, we can just re-use what has been defined in STAC.

**Comment by soxofaan** [comment]
just a quick note (and I haven't checked if this is some kind of standardized extension) but the standard json module in python does support encoding NaN and infinity in JSON (by default):

```
>>> import json
>>> json.dumps([1,2,3, float("nan"), float("inf")])
'[1, 2, 3, NaN, Infinity]'
```

**Comment by m-mohr** [comment]
That's unfortunately a Python-specific thing and not supported in other environments. In JS (and most other languages) this will lead to an error.

**Comment by m-mohr** [comment]
Once this is done, update the implementation guide (chapter 'log process') correspondingly.
